### PR TITLE
add delegation data structure and methods

### PR DESCRIPTION
**Pull request by chaosma**
- delegation data structure and methods
- validator data structure and methods
- commission data structure and methods

### CODE REVIEW

File: staking/types/delegation.go [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+      Balance:                   %s
+`,
+			i, entry.CreationHeight, entry.CompletionTime, entry.InitialBalance, entry.SharesDst, entry.Balance,
+		)
+	}
+
+	return strings.TrimRight(out, "\n")
+}
+
+type redelegationRespAlias RedelegationResponse
```
Comment (fxfactorial) [inline staking/types/delegation.go]: technically alias in go lang is of form: `type f = g`

File: staking/types/validator.go [hunk 2]
```diff
@@ -5,6 +5,16 @@ import (
...
+)
+
+// Define validator staking related const
+const (
+	MaxNameLength            = 70
+	MaxIdentityLength        = 3000
+	MaxWebsiteLength         = 140
+	MaxSecurityContactLength = 140
+	MaxDetailsLength         = 280
 )
```
Comment (fxfactorial) [inline staking/types/validator.go]: How you picked these magic numbers

File: staking/types/delegation.go (hunk 1 above)
Comment (chaosma) [reply staking/types/delegation.go]: this is removed from latest commit

File: staking/types/validator.go (hunk 2 above)
Comment (chaosma) [reply staking/types/validator.go]: I copied these numbers, we can always modify it later. This is just a start

**Pull request merged**